R/utils.R

```diff
@@ -74,3 +74,113 @@ write_test <- function(
     , file
   )
 }
+
+
+
+
+new_test_data <- function(numrow = 10){
+  tibble(
+
+    fld_factor = factor(sample(c("a","b","c"),size=numrow, replace=TRUE), levels=c("a","b","c"))
+
+    , fld_datetime = paste(
+      new_date_col(numrow)
+      , new_time_col(numrow)
+    )
+    , fld_date = new_date_col(numrow)
+    , fld_time = new_time_col(numrow)
+
+    , fld_binary = sample(c(1,0), numrow, TRUE)
+
+    , fld_integer = new_integer_col(numrow)
+    , fld_integer_alt = new_integer_col(numrow)
+
+    , fld_double = runif(numrow, -10000, 10000)
+    , fld_double_alt = new_double_col(numrow)
+
+    , fld_logical = new_logical_col(numrow)
+    , fld_logical_alt = new_logical_col(numrow)
+
+    , fld_character = new_character_col(numrow = numrow)
+    , fld_character_symbol = new_character_col(
+      numrow
+      , strsplit("~!@#$%^&*(){}|[]\\;'./,?><-_==+","")[[1]]
+      )
+    , fld_character_lead_trail_whitespace = paste0(
+      new_character_col(numrow, " ", 7)
+      , new_character_col( numrow , maxlength = 10)
+      , new_character_col(numrow, " ", 7)
+    )
+
+    , fld_datetime_utc = paste(
+      new_date_col(numrow)
+      , new_time_col(numrow)
+      , "UTC"
+    )
+  ) %>% lapply(na_portion) %>% as_tibble()
+}
+
+new_integer_col <- function(numrow = 10, magnitude=10000) {
+  as.integer(sample(-magnitude:magnitude, numrow, TRUE))
+}
+
+new_double_col <- function(numrow = 10, magnitude = 10000) {
+  runif(numrow, -magnitude, magnitude)
+}
+
+new_date_col <- function(numrow = 10) {
+  paste(
+    sprintf(sample(1:12,numrow,TRUE), fmt="%02d")
+    ,sprintf(sample(1:28,numrow,TRUE), fmt="%02d")
+    ,sample(2000:2020,numrow,TRUE)
```

Comment:
Maybe 1990:2020? That way we can test stuff that crosses may break 2 digit conversions